src/Miso/Internal.hs

```diff
@@ -122,10 +122,10 @@ data Hydrate
   deriving (Show, Eq)
 -----------------------------------------------------------------------------
 -- | @Component@ state, data associated with the lifetime of a @Component@
-data ComponentState model action
+data ComponentState key model action
```

Comment:
I think the key type param should be explained at the minimum in the ComponentState haddocks.
There's a high risk of people confusing this particular key param with the "Key" used for speeding up vdom diffing.

Reply:
I will in a latter pass, but also this isn't a part of the public API